src/cogent3/core/annotation_db.py

```diff
@@ -998,25 +998,23 @@ def _update_db_from_other_db(
 
         table_names = other_db.table_names
 
-        with other_db.db:
-            for table_name in table_names:
-                sql, values = _select_records_sql(
-                    table_name=table_name, conditions={"seqid": seqids}
-                )
-
-                cursor = other_db._execute_sql(sql, values)
-
-                with self.db:
-                    target_cursor = self.db.cursor()
-                    columns = ", ".join([desc[0] for desc in cursor.description])
-                    placeholders = ", ".join(["?"] * len(cursor.description))
-                    insert_sql = (
-                        f"INSERT INTO {table_name} ({columns}) VALUES ({placeholders})"
-                    )
-
-                    target_cursor.executemany(insert_sql, cursor)
+        col_order = {
+            tname: [
+                row[1]
+                for row in other_db.db.execute(f"PRAGMA table_info({tname})").fetchall()
+            ]
+            for tname in table_names
+        }
 
-                    self.db.commit()  # not sure if this is needed...
+        for tname in other_db.table_names:
+            sql, vals = _select_records_sql(
+                table_name=tname, conditions={"seqid": seqids}
```

Comment:
Should we just assign the default value for `conditions` if there are no `seqids` passed in?  That is, before the loop
```python
conditions = {"seqid": seqids} if seqids else {}
```

Reply:
@GavinHuttley The funky handling of `seqid` input is to handle the edge case where `seqids=[None]`, because SQL treats None as a seqid. Not sure whether this is necessary to account for.

Reply:
I also think the declared type should be `OptionalStrList = typing.Optional[typing.Union[str, typing.List[str]]]`, or even `OptionalStrContainer = typing.Optional[typing.Union[str, typing.Sequence]]`